internal/proxy/connection/manager.go

```diff
@@ -52,19 +52,49 @@ func (s *connectionManager) checkLoop() {
 			return
 		case <-t.C:
 			s.removeLongInactiveClients()
+			// not sure if we should purge them periodically.
+			s.purgeIfNumOfClientsExceed()
 			t.Reset(paramtable.Get().ProxyCfg.ConnectionCheckIntervalSeconds.GetAsDuration(time.Second))
 		}
 	}
 }
 
+func (s *connectionManager) purgeIfNumOfClientsExceed() {
+	if int64(s.clientInfos.Len()) < paramtable.Get().ProxyCfg.MaxConnectionNum.GetAsInt64() {
+		return
+	}
+
+	log.Info("number of client infos exceed limit, purge the oldest",
+		zap.Int64("num", int64(s.clientInfos.Len())),
+		zap.Int64("limit", paramtable.Get().ProxyCfg.MaxConnectionNum.GetAsInt64()))
+
+	diffNum := int64(s.clientInfos.Len()) - paramtable.Get().ProxyCfg.MaxConnectionNum.GetAsInt64()
```

Comment:
what if paramtable.Get().ProxyCfg.MaxConnectionNum.GetAsInt64() changed?

Reply:
the worst case is that we purged more client infos, which i think is not very important.